### ISSUE TITLE
`hardhat_metadata` network method returns incorrect fork information

### ISSUE THREAD

**Issue by ItsNickBarry** [NomicFoundation/edr#366]
### Version of Hardhat

2.22.2

### What happened?

`hardhat_metadata` indicates that the chain id of any forked network is 31337, identical to the hardhat network itself.  This occurs when using `hardhat_reset` and when configuring a fork via the config.

If there's any other way to get the chain id of the forked network, please let me know.

### Minimal reproduction steps

```
await hre.network.provider.send('hardhat_reset', [{ forking: { jsonRpcUrl: 'https://eth.llamarpc.com' }}]);
await hre.network.provider.send('hardhat_metadata', []);
```

**Comment by Wodann** [comment]
We released [EDR v0.3.5](https://www.npmjs.com/package/@nomicfoundation/edr/v/0.3.5) with a fix for this issue.

If you’re using the latest Hardhat release, in order to use EDR v0.3.4, you can remove your `node_modules` directory and npm/yarn/pnpm lock file and reinstall your dependencies. You can verify that you’re using the correct version of EDR by running the following command: 

`npm ls @nomicfoundation/edr `

It should print `0.3.5`.

**Comment by ItsNickBarry** [comment]
Working now, thanks.
